src/jobs/keys-update/test/detect-reorg.spec.ts

```diff
@@ -0,0 +1,173 @@
+  let executionProviderService: ExecutionProviderService;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        {
+          provide: LOGGER_PROVIDER,
+          useValue: {
+            log: jest.fn(),
+          },
+        },
+        {
+          provide: StakingRouterService,
+          useValue: {
+            getStakingRouterModuleImpl: () => ({
+              getCurrentNonce() {
+                return 1;
+              },
+            }),
+          },
+        },
+        {
+          provide: ElMetaStorageService,
+          useValue: {
+            update: jest.fn(),
+          },
+        },
+        {
+          provide: ExecutionProviderService,
+          useValue: {
+            getFullBlock: jest.fn(),
+          },
+        },
+        {
+          provide: SRModuleStorageService,
+          useValue: {
+            findOneById: jest.fn(),
+            upsert: jest.fn(),
+          },
+        },
+        StakingModuleUpdaterService,
+      ],
+    }).compile();
+
+    updaterService = module.get<StakingModuleUpdaterService>(StakingModuleUpdaterService);
+    executionProviderService = module.get<ExecutionProviderService>(ExecutionProviderService);
+  });
+
+  it('should be defined', () => {
+    expect(updaterService).toBeDefined();
+  });
+
+  it('parent hash of the currentBlock matches the hash of the prevBlock', async () => {
+    const updaterState: UpdaterState = {
+      lastChangedBlockHash: '0x1',
+      isReorgDetected: false,
+    };
+
+    jest
+      .spyOn(executionProviderService, 'getFullBlock')
+      .mockImplementationOnce(async () => ({ number: 2, hash: '0x2', timestamp: 1, parentHash: '0x1' } as any));
+
+    jest
+      .spyOn(executionProviderService, 'getFullBlock')
+      .mockImplementationOnce(async () => ({ number: 1, hash: '0x1', timestamp: 1, parentHash: '0x0' } as any));
+
+    expect(await updaterService.isReorgDetected(updaterState, '0x1', '0x1')).toBeFalsy();
+    expect(updaterState.isReorgDetected).toBeFalsy();
+  });
+
+  it('parent hash of the currentBlock matches the hash of the prevBlock', async () => {
+    const updaterState: UpdaterState = {
+      lastChangedBlockHash: '0x1',
+      isReorgDetected: false,
+    };
+
+    jest
+      .spyOn(executionProviderService, 'getFullBlock')
+      .mockImplementationOnce(async () => ({ number: 2, hash: '0x2', timestamp: 1, parentHash: '0x1' } as any));
+
+    jest
+      .spyOn(executionProviderService, 'getFullBlock')
+      .mockImplementationOnce(async () => ({ number: 1, hash: '0x1', timestamp: 1, parentHash: '0x0' } as any));
+
+    expect(await updaterService.isReorgDetected(updaterState, '0x1', '0x1')).toBeFalsy();
+    expect(updaterState.isReorgDetected).toBeFalsy();
+  });
+
+  it('same block number but different hashes', async () => {
+    const updaterState: UpdaterState = {
+      lastChangedBlockHash: '0x1',
+      isReorgDetected: false,
+    };
+
+    jest
```

Comment:
isReorgDetected(updaterState: UpdaterState, prevBlockHash: string, currentBlockHash: string)

I think it should be mock for these calls:
   const currentBlock = await this.executionProvider.getFullBlock(currentBlockHash);
   const prevBlock = await this.executionProvider.getFullBlock(prevBlockHash);
   
   we requested by '0x2' but got '0x1'

Reply:
fixed